clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -268,6 +268,9 @@ private KafkaChannel buildChannel(SocketChannel socketChannel, String id, Select
         }
         key.attach(channel);
         this.channels.put(id, channel);
+        // Include PLAINTEXT in successful authentication metric, all others added after KafkaChannel.prepare()
+        if (channel.ready())
+            sensors.successfulAuthentication.record();
```

Comment:
Would it be better to do this after `finishConnect` in `pollSelectionkeys`? Alternatively, do we want to track authentications for `PLAINTEXT`? Maybe that's not helpful.

Reply:
I wasn't intending to track authentications for `PLAINTEXT` when I wrote the KIP. But while implementing, I thought we are tracking unauthenticated SSL, so we probably ought to track PLAINTEXT too for consistency. Since the values are tracked at listener level, it should be ok?